### PR TITLE
model clean method

### PR DESCRIPTION
**Pull request by jan-www**
Provide model clean method and parameter reset method for tensor memory release

### CODE REVIEW

File: src/model/base.ts [hunk 1]
```diff
@@ -1,12 +1,82 @@
...
 
+export abstract class BaseEstimator {
+  public estimatorType: string;
+  public nFeature: number;
+
+  /**
+   * Check if input feature match the required feature size. If reset is true,
+   * reset nFeature = x.shape[1]
+   *
+   * @param x Input feature of shape (nSample, nFeatures).
```
Comment (yorkie) [inline src/model/base.ts:21]:
```suggestion
   * @param input feature of shape (nSample, nFeatures).
```

In tsdoc, we don't need to specify the variable name.

File: src/model/base.ts [hunk 2]
```diff
@@ -1,12 +1,82 @@
...
+   * @param reset if true, the `nFeatures` attribute is set to `x.shape[1]`.
```
Comment (yorkie) [inline src/model/base.ts:22]:
```suggestion
   * @param if true, the `nFeatures` attribute is set to `x.shape[1]`.
```
Ditto.

File: src/model/base.ts [hunk 3]
```diff
@@ -1,12 +1,82 @@
...
+   */
+  public checkAndSetNFeatures(x: Tensor, reset: boolean): void {
+    if (x?.shape?.length !== 2) {
+      throw new TypeError('Input should be 2D tensor');
```
Comment (yorkie) [inline src/model/base.ts:26]:
```suggestion
      throw new TypeError('input should be 2D tensor');
```
I just found other messages are in lower-case for the first char.

File: src/model/base.ts [hunk 4]
```diff
@@ -1,12 +1,82 @@
...
+      }
+    }
+  }
+  /**
+   * Reset given parameter. If type of parameter is tensor, previous value
+   * will be disposed.
+   * @param paramKey parameter name
+   * @param val value to be set
+   */
+  public cleanAndSetParam(paramKey: Extract<keyof this, string>, val: any): void {
```
Comment (yorkie) [inline src/model/base.ts:50]: See GitHub Actions' comment.

File: test/node/model/base.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+
+import { BaseEstimator } from '../../../src/model/base';
+import * as tf from '@tensorflow/tfjs-core';
+import { assert } from 'chai';
+import 'mocha';
```
Comment (yorkie) [inline test/node/model/base.ts:5]: Is this required?

File: test/node/model/base.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class TestEstimator extends BaseEstimator {
+  public a: tf.Tensor;
+  public b: tf.Tensor;
+  public c: number;
```
Comment (yorkie) [inline test/node/model/base.ts:10]: The above 2 members are useless.

**Review by yorkie — CHANGES_REQUESTED** [review]

File: test/node/model/base.ts (hunk 5 above)
Comment (jan-www) [reply test/node/model/base.ts:5]: what do you mean?

Comment (yorkie) [reply test/node/model/base.ts:5]: This line might be removed.

**Review by yorkie — APPROVED** [review]

File: src/model/base.ts [hunk 7]
```diff
@@ -1,12 +1,85 @@
...
+  /**
+   * Check if input feature match the required feature size. If reset is true,
+   * reset nFeature = x.shape[1]
+   *
+   * @param x input feature of shape (nSample, nFeatures).
+   * @param reset if true, the `nFeatures` attribute is set to `x.shape[1]`.
+   */
+  public checkAndSetNFeatures(x: Tensor, reset: boolean): void {
+    if (x?.shape?.length !== 2) {
+      throw new TypeError('Input should be 2D tensor');
```
Comment (FeelyChau) [inline src/model/base.ts:26]: Is it better to use type Tensor2d?

**Review by FeelyChau — CHANGES_REQUESTED** [review]

File: src/model/base.ts [hunk 8]
```diff
@@ -1,12 +1,85 @@
...
+
+export class BaseClustering extends BaseEstimator {
+  constructor() {
+    super();
+    this.estimatorType = 'clustering';
+  }
+  /**
+   * Validate the input of clustering task
+   * @param x input data
+   * @returns tensor of input data
```
Comment (FeelyChau) [inline src/model/base.ts:72]: Missing parameter `reset`.

**Review by FeelyChau — CHANGES_REQUESTED** [review]

**Review by yorkie — APPROVED** [review]
LGTM

**Pull request merged**